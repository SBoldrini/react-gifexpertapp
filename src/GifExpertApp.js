import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div className="container">
      <div className='row'>
        <div className='col-6'>
          <h1 className="text-gradient">Search all the Gifs</h1>
          <AddCategory setCategories={ setCategories }/>
        </div>
      </div>

      <div className='row'>
        <div className='col-12'>
          {
            categories.map((category) => (
              <GifGrid
                key={category}
                category={category}

              />
            ))
          }
          
        </div>
      </div>
    </div>
  );
}

export default GifExpertApp;


