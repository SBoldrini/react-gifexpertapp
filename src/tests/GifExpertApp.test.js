import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Prueba del componente <GifExpertApp />', () => {
  
  test('Debe de mostrarse correctamente', () => {

    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
    
  });


  test('Debe de mostra una lista de categorÃ­as', () => {

    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
    
  });
  
  

});
