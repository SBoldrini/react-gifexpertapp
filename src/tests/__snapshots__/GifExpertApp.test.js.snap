// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifExpertApp /> Debe de mostra una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Prueba del componente <GifExpertApp /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
